### DISCUSSION TITLE
Any way to disable captcha on local dev ?

### DISCUSSION THREAD

**Discussion by hakib** [django-recaptcha/django-recaptcha#252] [Q&A]
In my settings 

```
NOCAPTCHA = True
RECAPTCHA_USE_SSL = True
RECAPTCHA_PUBLIC_KEY = 'recaptcha-public-key'
RECAPTCHA_PRIVATE_KEY = 'recaptcha-private-key'
os.environ['RECAPTCHA_TESTING'] = 'True'
```

When rendering the field I get the captcha box with `ERROR: Invalid site key`.

I also tried suplying the initial fields to the form 

```
initial['g-recaptcha-response'] = 'PASSED'
initial['recaptcha_response_field'] = 'PASSED'
```

Same result. 

Idealy it should be possible to work on local dev env with no internet connection meaning under `DEBUG=True` or some `RECAPTCHA_DISABLED=True` the template will not even render and the field will always be valid.

Am I missing something?

Thanks, Haki.

**Comment by vstoykov** [comment]
In order recaptcha to be rendered it uses google servers to communicate. For that reason you need working internet connection.

If you want to develop without internet you just need to make your forms to not include captcha field when debug (but by this way you cant test if your configuration is correct), or to use another application for captcha validation which generates images and challenges locally (like django-simple-captcha).

**Comment by barseghyanartur** [comment]
As far as I can see, it should work as documented.
- [Read the docs](https://github.com/praekelt/django-recaptcha#unit-testing)
- [Tests](https://github.com/praekelt/django-recaptcha/blob/develop/captcha/tests.py#L16)
- [Code](https://github.com/praekelt/django-recaptcha/blob/develop/captcha/fields.py#L63)

The only thing I doubt is perhaps instead of providing the "PASSED" in initial you should instead provide it's value in data:

```
import os

from django.forms import Form
from captcha import fields

os.environ['RECAPTCHA_TESTING'] = 'True'

class TestForm(Form):
    captcha = fields.ReCaptchaField(attrs={'theme': 'white'})

form_params = {'recaptcha_response_field': 'PASSED'}
form = TestForm(form_params)

assert(form.is_valid())
```

**Comment by marcusamargi** [comment]
I am unable to disable on dev environment.

I don't print the "captcha" field to the template because I don't want to show it on dev.

For me, in `catcha/fields.py` the parameter `values` contains [None, None] and throws an exception.

```
def clean(self, values):
    super(ReCaptchaField, self).clean(values[1])
    recaptcha_challenge_value = smart_unicode(values[0])
    recaptcha_response_value = smart_unicode(values[1])

    if os.environ.get('RECAPTCHA_TESTING', None) == 'True' and \
            recaptcha_response_value == 'PASSED':
        return values[0]
```
